app/components/publications/overview/reports/generate-report-modal.js

```diff
@@ -0,0 +1,374 @@
+  //    if: a mandatee resigned in the year before the publication
+  //  2. a person can be selected person that was not a mandatee in that year
+  //    if a person is selected
+  //      and then the publication year is changed
+  @tracked mandateePersons;
+  @tracked selectedMandateePersons = [];
+
+  @tracked governmentDomains;
+  @tracked selectedGovernmentDomains = [];
+
+  @tracked regulationTypes;
+  @tracked selectedRegulationTypes = [];
+
+  constructor() {
+    super(...arguments);
+
+    const currentYear = new Date().getFullYear();
+    if (this.args.userInputFields.publicationYear) {
+      this.publicationYear = currentYear;
+    }
+
+    if (this.args.userInputFields.decisionDateRange) {
+      this.decisionDateRangeStart = new Date(currentYear, 0, 1, 0, 0, 0, 0);
+      this.decisionDateRangeEnd = undefined;
+    }
+
+    if (this.args.userInputFields.governmentDomains) {
+      this.loadGovernmentDomains.perform();
+    }
+
+    if (this.args.userInputFields.regulationTypes) {
+      this.loadRegulationTypes.perform();
+    }
+  }
+
+  get isLoading() {
+    return (
+      this.loadGovernmentDomains.isRunning || this.loadRegulationTypes.isRunning
+    );
+  }
+
+  get isDecisionDateRangeStartValid() {
+    if (this.decisionDateRangeStart === undefined) return false;
+    if (!this.isWithinSensibleRange(this.decisionDateRangeStart)) return false;
+    return this.isDecisionDateRangeValid;
+  }
+
+  get isDecisionDateRangeEndValid() {
+    if (this.decisionDateRangeEnd === undefined) return true;
+    if (!this.isWithinSensibleRange(this.decisionDateRangeEnd)) return false;
+    return this.isDecisionDateRangeValid;
+  }
+
+  isWithinSensibleRange(date) {
+    const minDate = new Date(1981, 1, 0, 0, 0, 0, 0);
+    const currentDate = new Date();
+    return minDate <= date && date <= currentDate;
+  }
+
+  get isDecisionDateRangeValid() {
+    const canValidateRange =
+      this.decisionDateRangeStart !== undefined &&
+      this.decisionDateRangeEnd !== undefined;
+    // return true: no range => so skip this step
+    if (!canValidateRange) return true;
+    return this.decisionDateRangeStart < this.decisionDateRangeEnd;
+  }
+
+  // using getters and setters to transform the year from a string to a number
+  // Input uses the property bound to @value to get and set the value
+  // It sets it to a string of the inputted number
+  // Storing it in the component as a number seems more understandable and usable
+  get publicationYearAsString() {
+    return this.publicationYear;
+  }
+
+  set publicationYearAsString(value) {
+    if (value === '') {
+      this.publicationYear = undefined;
+    } else {
+      // necessary to convert string value of <Input /> (even with @type="number")
+      const number = Number.parseInt(value);
+      if (!Number.isNaN(number)) {
+        this.publicationYear = number;
+      }
+    }
+  }
+
+  /// TODO: try out html min and max attributes and required
```

Comment:
Was this tried out? If so, you can remove this TODO comment

Reply:
Het leek me niet meer de moeite om het uit te testen. Ik heb deze comment laten staan, mochten jullie het anders zien.

Reply:
Het was me niet duidelijk of het gewenst was, maar heb dit geprobeerd. Het werkt, maar dwingt niks af bij typ-invoer, dus de validaties staan er ook nog. Hopelijk is dit het laatste dan.